src/Session.js

```diff
@@ -612,15 +613,12 @@ Session.prototype = {
        eventHandlers = options.eventHandlers || {},
        mangle = options.mangle || null;
 
-    if (eventHandlers.succeeded) {
-      this.reinviteSucceeded = eventHandlers.succeeded;
-    } else {
-      this.reinviteSucceeded = function(){
-        SIP.Timers.clearTimeout(self.timers.ackTimer);
-        SIP.Timers.clearTimeout(self.timers.invite2xxTimer);
-        self.status = C.STATUS_CONFIRMED;
-      };
-    }
+    this.reinviteSucceeded = function(){
+      SIP.Timers.clearTimeout(self.timers.ackTimer);
+      SIP.Timers.clearTimeout(self.timers.invite2xxTimer);
+      self.status = C.STATUS_CONFIRMED;
+      eventHandlers.succeeded && eventHandlers.succeeded.apply(this, arguments);
```

Comment:
`eventHandlers.succeeded` should be pull off of the eventHandlers object immediately when `sendReinvite()` is called.  Otherwise, we run into issues when people reuse `options` objects across multiple calls.